File: java-engine/src/main/java/hu/elteik/knowledgelab/javaengine/app/dto/SimulationStep.java
package hu.elteik.knowledgelab.javaengine.app.dto;

import java.util.List;

import hu.elteik.knowledgelab.javaengine.core.models.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SimulationStep<NodeType extends Node, EdgeType extends Edge, RobotType extends Robot> {

    // Meta
    private AlgorithmType algorithmType;
    private SimulationState simulationState;
    private Long step;

    // Model(s)
    private Graph<NodeType, EdgeType> graph;
    private List<RobotType> robotList;
    
}

File: dispersion-app/src/app/models/dto/SimulationStep.ts
import { Edge } from "../core/Edge";
import { Graph } from "../core/Graph";
import { Node } from "../core/Node";
import { Robot } from "../core/Robot";
import { AlgorithmType } from "../utils/AlgorithmType";
import { SimulationState } from "../utils/SimulationState";

export class SimulationStep<NodeType extends Node, EdgeType extends Edge, RobotType extends Robot> {

    algorithmType: AlgorithmType;
    simulationState: SimulationState;
    step: number;
    graph: Graph<NodeType, EdgeType>;
    robotList: RobotType[];

    constructor(algorithmType?: AlgorithmType, simulationState?: SimulationState, step?: number, graph?: Graph<NodeType, EdgeType>, robotList?: RobotType[]) {
        this.algorithmType = algorithmType;
        this.simulationState = simulationState;
        this.step = step;
        this.graph = graph;
        this.robotList = robotList;
    }

}